hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -528,6 +528,11 @@ private static boolean checkUnsupportedAttributeNaming(TypeDescription managedCt
 		// Check name of the getter/setter method with persistence annotation and getter/setter method name that doesn't refer to an entity field
 		// and will return false.  If the property accessor method(s) are named to match the field name(s), return true.
 		final AccessType defaultAccessType = determineDefaultAccessType( managedCtClass );
+
+		if ( AccessType.PROPERTY == defaultAccessType) {
+			return true;
+		}
```

Comment:
Not sure if you're done yet, but just in case: this will short-circuit the strategy selection below (see the switch) and skip logging. You probably want something more involved than an early return?